utils/hct/hctdb.py

```diff
@@ -3245,8 +3257,77 @@ def add_attr_arg(title_name, scope, args, doc):
         add_attr_arg("Unroll", "l", "Unroll the loop until it stops executing or a max count", [{"name":"Count", "type":"int"}])
         self.attributes = attributes
 
+class InstructionSignature:
+    def __init__(self, fn_attr, ret_type, name, args):
+        self.fn_attr = fn_attr
+        self.ret_type = ret_type
+        self.name = name
+        self.args = args
+    def __str__(self):
+        str_args = ",".join(self.args)
+        return "[{}] {} {}({})".format(self.fn_attr, self.ret_type, self.name, str_args)
+
+# user defined function
+# return false on instructions that should fail the query
+def query(insts, inst):
+    #ex 
+    """
+    if inst.ret_type == "v":
+        return true
+    return false
+    """
+
+    #ex 
+    """
+    if inst.fn_attr == "" and "Quad" in inst.name:
+        found = false
+        for other_inst in insts:
+            if not (other_inst == inst) and "Quad" in other_inst.name and other_inst.fn_attr != ""
+                found = true
+                break
+        return found
```

Comment:
The way this is called now, I don't know why you would check all the other instructions if the one you are on failed. The calling function for this iterates through all the functions so each should get a try? 

In which case, I don't know if the insts parameter is needed

Reply:
The insts parameter will be useful if we want to query for instructions that should only pass based on the existence of other instructions. For example, querying for any instruction where there exists another instruction with a different function attribute but with all other properties the same.